pkg/eventcollector/listener.go

```diff
@@ -292,16 +292,21 @@ func (c *EventCollector) handleEvent(payload interface{}) interface{} {
 	}
 
 	// We need to get past newsroom events for the newsroom contract of a newly added watcher
-	if event.EventType() == "Application" {
-		newsroomAddr := event.EventPayload()["ListingAddress"].(common.Address)
+	if event.EventType() == "Application" || (event.ContractName() == "NewsroomFactory" && event.EventType() == "Instantiation") {
```

Comment:
Minor and prob can be done later, but these little checks could be wrapped in helper methods to make this more concise and readable.

Ex:
```
if c.isTcrApplicationEvent(event) || c.isNewsroomFactoryInstantiationEvent(event) {
    ...
}

...

func (...) isNewsroomFactoryInstantiationEvent(event ...) bool {
     return (event.ContractName() == "NewsroomFactory" && event.EventType() == "Instantiation")
}

func (...) isTcrApplicationEvent(event ...) bool {
     return (event.ContractName() == "CivilTCR" && event.EventType() == "Application")
}

```